src/tests/ftest/container/full_pool_container_create.py

```diff
@@ -34,84 +58,67 @@ def test_no_space_cont_create(self):
+                except DaosTestError as excep:
+                    if not err in repr(excep):
                         self.log.error("caught exception while writing "
                                        "object: %s", repr(excep))
-                        cont.close()
+                        self.container.close()
                         self.fail("caught exception while writing "
                                   "object: {}".format(repr(excep)))
                     else:
                         self.log.info("pool is too full for %s byte "
                                       "objects", obj_sz)
                         break
 
-        self.log.info("closing container")
-        cont.close()
-
         # query the pool
         self.log.info("Pool Query after filling")
         self.pool.set_query_data()
         self.log.info(
             "Pool %s query data: %s\n", self.pool.uuid, self.pool.query_data)
 
-        # create a 2nd container now that pool is full
-        try:
-            self.log.info("creating 2nd container")
-            cont2 = DaosContainer(self.context)
-            cont2.create(self.pool.pool.handle)
-            self.log.info("created 2nd container")
-
-            self.log.info("opening container 2")
-            cont2.open()
-
-            self.log.info("writing one more object, write expected to fail")
-            cont2.write_an_obj(my_str, my_str_sz, dkey, akey,
-                               obj_cls="OC_SX")
-            self.log.info("closing container")
-            cont2.close()
-            self.fail("wrote one more object after pool was completely filled,"
-                      " this should never print")
-        except DaosApiError as excep:
-            if not (err in repr(excep) or err2 in repr(excep)):
-                self.log.error("caught unexpected exception while "
-                               "writing object: %s", repr(excep))
-                self.log.info("closing container")
-                cont2.close()
-                self.fail("caught unexpected exception while writing "
-                          "object: {}".format(repr(excep)))
-            else:
-                self.log.info("correctly caught -1007 while attempting "
-                              "to write object in full pool")
-                self.log.info("closing container")
-                cont2.close()
+        # destroy container
+        self.container.destroy()
+
+        # check for free space to be returned back once aggregation is complete
+        # checking for a closer returned space value instead of exact value
+        # as the test is using scm only
+        counter = 1
+        threshold_value = free_space_before - (free_space_before *
+                                               threshold_percent)
+        free_space = self.pool.get_pool_free_space()
+        while free_space < threshold_value:
+            # try to wait for 4 x 30 secs for aggregation to be completed or
```

Comment:
So we are waiting for two minutes? Is this acceptable in in real world scenario?
What happens if there is constant IO's happening during this time? Probably, those will get -1007. Is it not?
Assume that IO is happening, can aggregation still run during this time?

Reply:
We are not exactly waiting for two minutes, we are waiting for maximum of up to two minutes. The first check will happen immediately, if aggregation hasn't reclaimed the space we wait for 30 secs and check again and repeat 4 more times. And if we still don't get the space back we fail the test (same thing as we are doing in other aggregation tests).
And yes aggregation doesn't return the space back immediately unless it is very small space. So even in real world scenario if somebody is playing with large amount of space they might have to wait a few seconds after removing/deleting the data.
And regarding constant IO, here we are only extracting the free space by querying, so even if it is full we should not get any error -1007 or anything else (if we do, something is broken).
Yes aggregation is always on by default and should be working in the background even if there is constant IO happening.

Reply:
yes, I didn't mean that we are waiting for 2 minutes... We are checking every 30 secs. I am not sure how fast aggregation reclaims the space. hmm. I was thinking aggregation has low priority and should not affect (or run during) IO. I maybe wrong.